File: clientandroid/app/src/main/java/com/hairdresser/pyshankov/FragmentFactory.java
package com.hairdresser.pyshankov;

import android.support.v4.app.Fragment;

import com.hairdresser.pyshankov.fragment.LogInFragment;
import com.hairdresser.pyshankov.fragment.SearchFragment;

/**
 * Created by pyshankov on 1/29/17.
 */

public class FragmentFactory {

    private static LogInFragment logInFragment;

    private static SearchFragment searchFragment;

    public static Fragment getLogInFragment(){
        if(logInFragment == null)
            logInFragment = new LogInFragment();
        return logInFragment;
    }

    public static Fragment getSearchFragment(){
        if(searchFragment == null)
            searchFragment = new SearchFragment();
        return searchFragment;
    }

    public static Fragment getFragment(String name){
        if(name.equals(LogInFragment.NAME))
            return getLogInFragment();
        if(name.equals(SearchFragment.NAME))
            return getSearchFragment();
       return null;
    }

}

File: clientandroid/app/src/main/kotlin/com/hairdresser/pyshankov/FragmentFactory.kt
package com.hairdresser.pyshankov

import android.support.v4.app.Fragment

import com.hairdresser.pyshankov.fragment.LogInFragment
import com.hairdresser.pyshankov.fragment.ProfileFragment
import com.hairdresser.pyshankov.fragment.SearchFragment

/**
 * Created by pyshankov on 1/29/17.
 */

object FragmentFactory {

    private var logInFragment: LogInFragment? = null

    private var searchFragment: SearchFragment? = null

    private var profileFragment: ProfileFragment? =null

    fun getLogInFragment(): LogInFragment {
        if (logInFragment == null)
            logInFragment = LogInFragment()
        return logInFragment as LogInFragment
    }

    fun getProfileFragment(): ProfileFragment{
        if(profileFragment == null)
            profileFragment = ProfileFragment();
        return profileFragment as ProfileFragment;
    }

    fun getSearchFragment(): SearchFragment {
        if (searchFragment == null)
            searchFragment = SearchFragment()
        return searchFragment as SearchFragment
    }

    fun getFragment(name: String): Fragment? {
        if (name == LogInFragment.NAME)
            return getLogInFragment()
        if (name == SearchFragment.NAME)
            return getSearchFragment()
        if (name == ProfileFragment.NAME)
            return getProfileFragment()
        return null
    }

}
